crates/ursa-network/src/service.rs

```diff
@@ -676,7 +677,15 @@ where
 
                 println!("cosmos");
             }
-            NetworkCommand::Put { cid: _, sender: _ } => (),
+            NetworkCommand::Put { cid, sender } => {
+                let swarm = self.swarm.behaviour_mut();
+                for peer in &self.peers {
+                    swarm
+                        .request_response
+                        .send_request(peer, UrsaExchangeRequest(RequestType::CacheRequest(cid)));
```

Comment:
Are we handling this request or is that coming in a different PR?

Reply:
Yeah, that is coming in a different PR.